aeon/tests/test_all_estimators.py

```diff
@@ -1043,8 +1037,97 @@ def test_valid_estimator_tags(self, estimator_instance):
 class TestAllEstimators(BaseFixtureGenerator, QuickTester):
     """Package level tests for all aeon estimators, i.e., objects with fit."""
 
+    def test_raises_not_fitted_error(self, estimator_instance, scenario, method_nsc):
+        """Check exception raised for non-fit method calls to unfitted estimators.
+
+        Tries to run all methods in NON_STATE_CHANGING_METHODS with valid scenario,
+        but before fit has been called on the estimator.
+
+        This should raise a NotFittedError if correctly caught,
+        normally by a self.check_is_fitted() call in the method's boilerplate.
+
+        Raises
+        ------
+        Exception if NotFittedError is not raised by non-state changing method
+        """
+        # call methods without prior fitting and check that they raise NotFittedError
+        with pytest.raises(NotFittedError, match=r"has not been fitted"):
+            scenario.run(estimator_instance, method_sequence=[method_nsc])
+
+    def test_non_state_changing_method_contract(
+        self, estimator_instance, scenario, method_nsc
+    ):
+        """Check that non-state-changing methods behave as per interface contract.
+
+        Check the following contract on non-state-changing methods:
+        1. do not change state of the estimator, i.e., any attributes
+            (including hyper-parameters and fitted parameters)
+        2. expected output type of the method matches actual output type
+            - only for abstract BaseEstimator methods, common to all estimator scitypes
+            list of BaseEstimator methods tested: get_fitted_params
+            scitype specific method outputs are tested in TestAll[estimatortype] class
+        3. No state changes from calling non state changing methods
+        """
+        # No point testing get_fitted_params here, does not change state
+        if method_nsc == "get_fitted_params":
+            return None
+
+        estimator = estimator_instance
+        set_random_state(estimator)
+
+        _, args_after = scenario.run(
+            estimator, method_sequence=["fit"], return_args=True
+        )
+        fit_args_after = args_after[0]
+        fit_args_before = scenario.args["fit"]
+        assert deep_equals(
+            fit_args_before, fit_args_after
+        ), f"Estimator: {type(estimator)} has side effects on arguments of fit"
+
+        # dict_before = copy of dictionary of estimator before predict, post fit
+        dict_before = estimator.__dict__.copy()
+
+        # skip test if vectorization would be necessary and method predict_proba
+        # this is since vectorization is not implemented for predict_proba
+
```

Comment:
```suggestion
        # skip test if predict_proba is not implemented
```
Is it just me, or does this not make any sense in relation to all estimators (or maybe any)